addons/controller_icons/objects/ControllerIconTexture.gd

```diff
@@ -171,6 +171,9 @@ func _load_texture_path():
 	_textures = textures
 	_reload_resource()
 
+func _load_texture_path():
+	_load_texture_path_main_thread.call_deferred()
```

Comment:
Sadly we don't have access to the `call_thread_safe` [function](https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-method-call-thread-safe) as the icons don't inherit from Node; could you try if the following code works in your environment as well?
```suggestion
	# Ensure loading only occurs on the main thread
	if OS.get_thread_caller_id() != OS.get_main_thread_id():
		_load_texture_path_main_thread.call_deferred()
	else:
		_load_texture_path_main_thread()
```

Reply:
Yes, this works and is a much better solution 👍